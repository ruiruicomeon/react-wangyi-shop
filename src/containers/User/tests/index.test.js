import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import history from '@/utils/history';
import configureStore from '@/store';

import { User } from '../index';

describe('User 测试', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <User user={{ image: 'a', username: 'apple' }} />
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
