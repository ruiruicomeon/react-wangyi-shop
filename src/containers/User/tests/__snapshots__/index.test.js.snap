// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User 测试 snapshot 1`] = `
<div
  className="layout"
>
  <nav
    className="nav"
  >
    <div
      className="content container"
    >
      <div
        className="notice"
      >
        <img
          alt="speaker"
          className="notice__img"
          src="speaker.gif"
        />
        <ul
          className="notice__list"
        >
          <li
            className="notice__item"
          />
        </ul>
      </div>
      <div
        className="link"
      >
        <ul
          className="link__list"
        >
          <li
            className="link__item"
          >
            <div
              className="link__link"
              onClick={[Function]}
            >
              登录/注册
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link"
            >
              会员
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link"
            >
              甄选家
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link ant-dropdown-trigger"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              企业采购 
              <span
                aria-label="down"
                className="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link ant-dropdown-trigger"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              客户服务 
              <span
                aria-label="down"
                className="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </div>
          </li>
          <li
            className="link__item"
          >
            <div
              className="link__link"
            >
              <span
                className="link__appIcon"
              />
              APP
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="login"
    />
  </nav>
  <header
    className="header"
  >
    <div
      className=""
    >
      <div
        className="container content"
      >
        <a
          className="logo"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__img"
          />
        </a>
        <a
          className="smlogo"
          href="/"
          onClick={[Function]}
        >
          <span
            className="smlogo__img"
          />
        </a>
        <a
          className="cart ant-dropdown-trigger"
          href="/cart"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            className="cart__icon"
          />
          <span
            className="cart__title"
          >
            购物车
          </span>
          <span
            className="cart__num"
          >
            0
          </span>
        </a>
        <a
          className="user"
          href="/user"
          onClick={[Function]}
        >
          <span
            className="user__icon"
          />
        </a>
        <div
          className="search"
        >
          <div
            className="search__main"
          >
            <div
              className="search__wrap"
            >
              <div
                className="search__hide"
              >
                &gt;&gt;
              </div>
              <div
                className="search__prefix"
              />
              <input
                className="search__input"
                onChange={[Function]}
                placeholder="搜索"
                type="text"
                value=""
              />
            </div>
            <div
              className="search__btn"
            >
              <div
                className="search__icon"
              />
              <div
                className="search__text"
                onClick={[Function]}
              >
                搜索
              </div>
            </div>
          </div>
        </div>
        <div
          className="loading"
        >
          <div
            className="ant-spin ant-spin-spinning"
          >
            <span
              aria-label="loading"
              className="anticon anticon-loading ant-spin-dot"
              role="img"
              style={
                Object {
                  "fontSize": 24,
                }
              }
            >
              <svg
                aria-hidden="true"
                className="anticon-spin"
                data-icon="loading"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="container content"
  >
    <div
      className="breadcrumb ant-breadcrumb"
    >
      <span>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            className="breadcrumb__link"
            href="/"
            onClick={[Function]}
          >
            首页
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          &gt;
        </span>
      </span>
    </div>
    <div
      className="body"
    >
      <div
        className="sidebar sidebar"
      >
        <div
          className="category"
        >
          <div
            className="category__title"
          >
            账号管理
          </div>
          <ul
            className="category__list"
          >
            <li
              className="category__item category__item_active"
            >
              <a
                href="/user"
                onClick={[Function]}
              >
                个人中心
              </a>
            </li>
            <li
              className="category__item"
            >
              <a
                href="/user"
                onClick={[Function]}
              >
                消息通知
              </a>
            </li>
            <li
              className="category__item"
            >
              <a
                href="/user"
                onClick={[Function]}
              >
                账号信息
              </a>
            </li>
            <li
              className="category__item"
            >
              <a
                href="/user"
                onClick={[Function]}
              >
                地址管理
              </a>
            </li>
            <li
              className="category__item"
            >
              <a
                href="/user"
                onClick={[Function]}
              >
                账号安全
              </a>
            </li>
            <li
              className="category__item"
            >
              <a
                href="/user"
                onClick={[Function]}
              >
                我的积分
              </a>
            </li>
          </ul>
        </div>
        <div
          className="category"
        >
          <div
            className="category__title"
          >
            交易管理
          </div>
          <ul
            className="category__list"
          >
            <li
              className="category__item"
            >
              <a
                href="/orders"
                onClick={[Function]}
              >
                订单管理
              </a>
            </li>
            <li
              className="category__item"
            >
              <a
                href="/orders"
                onClick={[Function]}
              >
                优惠券
              </a>
            </li>
          </ul>
        </div>
        <div
          className="category"
        >
          <div
            className="category__title"
          >
            服务中心
          </div>
          <ul
            className="category__list"
          >
            <li
              className="category__item"
            >
              售后记录
            </li>
            <li
              className="category__item"
            >
              帮助中心
            </li>
          </ul>
        </div>
      </div>
      <div
        className="user"
      >
        <div
          className="header"
        >
          <div
            className="header__left"
          >
            <div
              className="info"
            >
              <img
                alt="avatar"
                className="info__img"
                src="a"
              />
              <div
                className="info__name"
              >
                apple
              </div>
            </div>
          </div>
          <div
            className="header__right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="intro"
    >
      <div
        className="container content"
      >
        <div
          className="intro__col"
        >
          <div
            className="intro__title"
          >
            客户服务
          </div>
          <div
            className="intro__content"
          >
            <div
              className="buttons"
            >
              <div
                className="button"
              >
                <div
                  className="button__service"
                />
                <div
                  className="button__title"
                >
                  在线客服
                </div>
              </div>
              <div
                className="button"
              >
                <div
                  className="button__feedback"
                />
                <div
                  className="button__title"
                >
                  用户反馈
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="intro__col"
        >
          <div
            className="intro__title"
          >
            何为严选
          </div>
          <div
            className="intro__content"
          >
            <div
              className="text"
            >
              网易原创生活类电商品牌，秉承网易一贯的严谨态度，我们深入世界各地，从源头全程严格把控商品生产环节，力求帮消费者甄选到优质的商品
            </div>
            <div
              className="follow"
            >
              <div>
                关注我们 :
              </div>
              <div
                className="follow__qq"
              />
              <div
                className="follow__weixin"
              />
              <div
                className="follow__weibo"
              />
            </div>
          </div>
        </div>
        <div
          className="intro__col"
        >
          <div
            className="intro__title"
          >
            扫码下载严选APP
          </div>
          <div
            className="intro__content"
          >
            <div
              className="qr"
            >
              <img
                alt="qrcode"
                className="qr__image"
                src="qr.png"
              />
              <div
                className="qr__title"
              >
                下载领大额新人红包
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="link"
    >
      <div
        className="container content"
      >
        <div
          className="ensure"
        >
          <div
            className="ensure__item"
          >
            <div
              className="ensure__yi"
            />
            <div
              className="ensure__title"
            >
              网易自营电商
            </div>
          </div>
          <div
            className="ensure__item"
          >
            <div
              className="ensure__tui"
            />
            <div
              className="ensure__title"
            >
              30天无忧退换货
            </div>
          </div>
          <div
            className="ensure__item"
          >
            <div
              className="ensure__bao"
            />
            <div
              className="ensure__title"
            >
              满88元免邮费
            </div>
          </div>
          <div
            className="ensure__item"
          >
            <div
              className="ensure__ping"
            />
            <div
              className="ensure__title"
            >
              品质保证
            </div>
          </div>
        </div>
        <ul
          className="list"
        >
          <li
            className="list__item"
          >
            关于我们
          </li>
          <li
            className="list__item"
          >
            帮助中心
          </li>
          <li
            className="list__item"
          >
            售后服务
          </li>
          <li
            className="list__item"
          >
            配送与验收
          </li>
          <li
            className="list__item"
          >
            商务合作
          </li>
          <li
            className="list__item"
          >
            企业采购
          </li>
          <li
            className="list__item"
          >
            开放平台
          </li>
          <li
            className="list__item"
          >
            搜索推荐
          </li>
          <li
            className="list__item"
          >
            友情链接
          </li>
          <li
            className="list__item"
          >
            廉正举报
          </li>
        </ul>
        <ul
          className="certs"
        >
          <li
            className="certs__item"
          >
            食品经营许可证：JY13301080111719
          </li>
          <li
            className="certs__item"
          >
            出版物经营许可证：新出发滨字第0132号
          </li>
          <li
            className="certs__item"
          >
            妙得ICP证号：ICP 证浙B2-20160106
          </li>
          <li
            className="certs__item"
          >
            浙杭食药监械经营备20171029号
          </li>
          <li
            className="certs__item"
          >
            营业执照
          </li>
          <li
            className="certs__item"
          >
            网易公司版权所有 © 1997-2020
          </li>
        </ul>
        <ul
          className="certs"
        >
          <li
            className="certs__item"
          >
            无线电发射设备销售备案：42201919712647
          </li>
          <li
            className="certs__item"
          >
            (浙杭）网械企备字[2019]第00119号
          </li>
          <li
            className="certs__item"
          >
            （粤）网械平台备字（2019）第00004号
          </li>
          <li
            className="certs__item"
          >
            单用途商业预付卡备案证：330100AAC0024
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
